src/Qir/check-sources-formatted.ps1

```diff
@@ -0,0 +1,31 @@
+#Requires -PSEdition Core
+#Requires -Version 7.1
+
+param (
+    [Parameter()]
+    [String]
+    $DirPath = "$PSScriptRoot"
+)
+
+$tmpFile = "format.log"
+
+$OldErrorActionPreference = $ErrorActionPreference
+$ErrorActionPreference='Continue'
+"$DirPath/*.cpp","$DirPath/*.c","$DirPath/*.h","$DirPath/*.hpp" | get-childitem -Recurse | %{$_.fullname -replace "\\", "/"} `
+    | ?{$_ -notlike "*/Externals/*"} | ?{$_ -notlike "*/drops/*"} | ?{$_ -notlike "*/bin/*"} `
+    | %{clang-format -n -style=file $_} 2>format.log
```

Comment:
A couple suggestions here:
- short-hand such as `?` and `%` is generally not advised in production scripts, since that shorthand can be overridden.
- can refactor out the join with `$DirPath` into a new `ForEach-Object` and use `Join-Path` for more robust join handling cross-platform
- can use `` ` `` to wrap lines for readability

```suggestion
"*.cpp", "*.c", "*.h", "*.hpp" `
    | ForEach-Object { Join-Path $DirPath $_ } `
    | Get-ChildItem -Recurse `
    <# clang-format accepts path names with Unix-style separators. #> `
    | ForEach-Object {$_.FullName -replace "\\", "/"} `
    | Where-Object { $_ -notlike "*/Externals/*" } `
    | Where-Object { $_ -notlike "*/drops/*" } `
    | Where-Object { $_ -notlike "*/bin/*" } `
    | ForEach-Object {
        clang-format -n -style=file $_
    } 2>format.log
```

Reply:
It also looks like `$tmpFile` isn't used here, but rather `format.log` is hard-coded?